econml/dml/dynamic_dml.py

```diff
@@ -0,0 +1,737 @@
+                    f"Expected a list of time period * d_t, instead got a treatment array of shape {T.shape}."
+                if self.transformer:
+                    T = np.hstack([
+                        base_expand_treatments(
+                            X, T[:, [kappa]])[1] for kappa in range(self._n_periods)
+                    ])
+            outTs.append(T)
+        return (X,) + tuple(outTs)
+
+    @property
+    def bias_part_of_coef(self):
+        return self.ortho_learner_model_final_._model_final._fit_cate_intercept
+
+    @property
+    def fit_cate_intercept_(self):
+        return self.ortho_learner_model_final_._model_final._fit_cate_intercept
+
+    @property
+    def original_featurizer(self):
+        # NOTE: important to use the rlearner_model_final_ attribute instead of the
+        #       attribute so that the trained featurizer will be passed through
+        return self.ortho_learner_model_final_._model_final_trained[0]._original_featurizer
+
+    @property
+    def featurizer_(self):
+        # NOTE This is used by the inference methods and has to be the overall featurizer. intended
+        # for internal use by the library
+        return self.ortho_learner_model_final_._model_final_trained[0]._featurizer
+
+    @property
+    def model_final_(self):
+        # NOTE This is used by the inference methods and is more for internal use to the library
+        #      We need to use the rlearner's copy to retain the information from fitting
+        return self.ortho_learner_model_final_.model_final_
+
+    @property
+    def model_final(self):
+        return self._gen_model_final()
+
+    @model_final.setter
+    def model_final(self, model):
+        if model is not None:
+            raise ValueError("Parameter `model_final` cannot be altered for this estimator!")
+
+    @property
+    def models_y(self):
+        return [[mdl._model_y for mdl in mdls] for mdls in super().models_nuisance_]
+
+    @property
+    def models_t(self):
+        return [[mdl._model_t for mdl in mdls] for mdls in super().models_nuisance_]
+
+    @property
+    def nuisance_scores_y(self):
+        return self.nuisance_scores_[0]
+
+    @property
+    def nuisance_scores_t(self):
+        return self.nuisance_scores_[1]
+
+    @property
+    def residuals_(self):
+        """
+        A tuple (y_res, T_res, X, W), of the residuals from the first stage estimation
+        along with the associated X and W. Samples are not guaranteed to be in the same
+        order as the input order.
+        """
+        if not hasattr(self, '_cached_values'):
+            raise AttributeError("Estimator is not fitted yet!")
+        if self._cached_values is None:
+            raise AttributeError("`fit` was called with `cache_values=False`. "
+                                 "Set to `True` to enable residual storage.")
+        Y_res, T_res = self._cached_values.nuisances
```

Comment:
residuals for Y and T are not correct here and it's complicated...we fit nuisance iteratively on different subgroup and it's not clear to me how to store all the nuisances but still inherit through `_OrthoLearner` we might need a dictionary format to store the residuals but that's not the case for the other classes.